retropie-shell-script-boilerplate.sh

```diff
@@ -52,13 +53,19 @@ function is_retropie() {
     [[ -d "$home/RetroPie" && -d "$home/.emulationstation" && -d "/opt/retropie" ]]
 }
 
-
+# If your script has dependencies, just use the DEPENDENCIES
+# variable on the definitions above. Otherwise, leave it as is.
 function check_dependencies() {
-    if ! which [COMMAND_TO_TEST] > /dev/null; then # (e.g if ! which git > /dev/null)
-        echo "ERROR: The [NAME_OF_THE_PACKAGE] package is not installed!" >&2 # (e.g. git)
-        echo "Please, install it with 'sudo apt-get install [NAME_OF_THE_PACKAGE]'." >&2 # (e.g. git)
-        exit 1
-    fi
+    local pkg
+    local err=0
+    for pkg in "${DEPENDENCIES[@]}"; do
+        if ! which "$pkg" &> /dev/null; then
+            echo "ERROR: The \"$pkg\" package is not installed!" >&2
```

Comment:
I try to avoid escaping characters (whenever possible), in this case the double quotes. I would just use single quotes. I think it's tedious to check for escaping characters in general :P 
What do you think? Is it a personal choice or a standard?

Reply:
neither personal choice nor standard, its necessary. Look:

```
$ pkg=package_name

$ echo "ERROR: The \"$pkg\" package is not installed!"
ERROR: The "package_name" package is not installed!

$ echo 'ERROR: The "$pkg" package is not installed!'
ERROR: The "$pkg" package is not installed!

```

See? When using single quotes the variables aren't expanded.

If you want more details about quoting variables in bash, take a look here:
http://tldp.org/LDP/abs/html/quotingvar.html

Reply:
I haven't explained myself properly. My question wasn't about single/double quotes in `echo`. I meant using escaped double quotes in `\"$pkg\"` instead of using single quotes `'$pkg'`, so the output would be:

```
$ pkg=package_name

$ echo "ERROR: The \"$pkg\" package is not installed!"
ERROR: The "package_name" package is not installed!

$ echo 'ERROR: The '$pkg' package is not installed!'
ERROR: The 'package_name' package is not installed!
```